code/json.sls

```diff
@@ -30,172 +30,185 @@
 
 (library (json)
   (export json-write
-	  json-read)
+          json-read)
   
   (import
-    (except (rnrs) define-record-type error)
-    (rnrs r5rs)
-    (rnrs mutable-pairs)
-    (only (srfi :1 lists) lset-union append-map fold)
-    (srfi :6 basic-string-ports)
-    (srfi :9 records)
-    (srfi :23 error)
-    (only (chezscheme) include)
-    (packrat))
+   (except (rnrs) define-record-type error)
+   (rnrs r5rs)
+   (rnrs mutable-pairs)
+   (only (srfi :1 lists) lset-union append-map fold iota)
+   (srfi :6 basic-string-ports)
+   (srfi :9 records)
+   (srfi :23 error)
+   (only (chezscheme) include)
+   (packrat))
 
   (define json-write
     (let ()
+      (define indent-level 0)
+      (define (display-level)
+        (display
+         (apply string-append (map (lambda (_) "  ") (iota indent-level)))))
       (define (write-ht vec p)
-	(display "{" p)
-	(do ((need-comma #f #t)
-	     (vec vec (cdr vec)))
-	    ((null? vec))
-	  (if need-comma
-	      (display ", " p)
-	      (set! need-comma #t))
-	  (let* ((entry (car vec))
-		 (k (car entry))
-		 (v (cdr entry)))
-	    (cond
-	      ((symbol? k) (write (symbol->string k) p))
-	      ((string? k) (write k p)) ;; for convenience
-	      (else (error "Invalid JSON table key in json-write" k)))
-	    (display ": " p)
-	    (write-any v p)))
-	(display "}" p))
+        (display "{\n" p)
+        (set! indent-level (+ indent-level 1))
+        (do ((need-comma #f #t)
+             (vec vec (cdr vec)))
+            ((null? vec))
+          (if need-comma
+              (display ",\n" p)
+              (set! need-comma #t))
+          (let* ((entry (car vec))
+                 (k (car entry))
+                 (v (cdr entry)))
+            (cond
+             ((symbol? k) (display-level) (display (symbol->string k) p))
```

Comment:
`(write (symbol->string k) p)` will fix the keys not being quoted. To match configlet you'll probably also want to convert to the `-` to `_`

Reply:
Cool, I figured that was it.